internal/pkg/manifest/validate.go

```diff
@@ -2000,6 +2031,80 @@ type validateARMOpts struct {
 	SpotFrom *int
 }
 
+func validateHealthCheckPorts(opts validateHealthCheckPortsOpts) error {
+	if opts.alb != nil {
+		for _, rule := range opts.alb.RoutingRules() {
+			// healthCheckPort is defined by RoutingRule.HealthCheck.Port, with fallback on RoutingRule.TargetPort, then image.port.
```

Comment:
is this logic to figure out the health check port defined elsewhere already? (maybe `package stack`?) if so, maybe we should put this `HealthCheckPort(imagePort) uint16` logic onto the routing rule 🤔

Reply:
Unfortunately the behavior isn't written in one place. Maybe this should be changed though 🤔 Currently the port is half determined by `stack/transformers.go` and half determined by the logic within the template. I think this can be easily changed though so there's exactly one determining function in the codebase for each TargetGroup health check port!

Reply:
I'm unsure what exactly to do about this now because I've realized that in order to change the `deploy/cloudformation/stack` package to use the function I've written in the `manifest` package as the source of truth for the HealthCheckPort, we would need to update some people's stack. Basically if they have the following manifest, the target group would undergo a change. 
```
nlb:
  port: 80
```
Would cause the target group to go from:
```
NLBTargetGroup:
  Port: 80
```
to:
```
NLBTargetGroup:
  HealthCheckPort: 80
```
This doesn't cause any change in behavior but requires a stack update. Maybe this is a change that should go along with updating the NLB/TargetGroup when the user specifies UDP?